pkg/subctl/cmd/validate_cni.go

```diff
@@ -78,4 +94,123 @@ func validateCNIInCluster(clusterName string, submariner *v1alpha1.Submariner) {
 		submariner.Status.NetworkPlugin)
 	status.QueueSuccessMessage(message)
 	status.End(cli.Success)
+
+	validateCalicoIPPoolsIfCalicoCNI(config)
+}
+
+func doesCalicoConfigMapExist(clientSet kubernetes.Interface) (*v1.ConfigMap, error) {
+	cmList, err := clientSet.CoreV1().ConfigMaps(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cm := range cmList.Items {
+		if cm.Name == "calico-config" {
+			return &cm, nil
+		}
+	}
+	return nil, nil
+}
+
+func validateCalicoIPPoolsIfCalicoCNI(config *rest.Config) {
+	dynClient, clientSet, err := getClients(config)
+	exitOnError("Error creating clients for cluster", err)
+
+	calicoConfig, _ := doesCalicoConfigMapExist(clientSet)
+	if calicoConfig == nil {
+		return
+	}
+
+	message := "Calico CNI detected, verifying if the Submariner IPPool pre-requisites are configured."
+	status.Start(message)
+
+	gateways := getGatewaysResource(config)
+	if gateways == nil {
+		message = "There are no gateways detected on the cluster"
+		status.QueueWarningMessage(message)
+		status.End(cli.Failure)
+		return
+	}
+
+	client := dynClient.Resource(calicoGVR)
+
+	ippoolList, err := client.List(metav1.ListOptions{})
+	if err != nil {
+		message := fmt.Sprintf("Error obtaining IPPools: %v", err)
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+		return
+	}
+
+	if len(ippoolList.Items) < 1 {
+		message := "Could not find any IPPools in the cluster"
+		status.QueueFailureMessage(message)
+		status.End(cli.Failure)
+		return
+	}
+
+	ippools := make(map[string]unstructured.Unstructured)
+	for _, pool := range ippoolList.Items {
+		cidr, found, err := unstructured.NestedString(pool.Object, "spec", "cidr")
+		if err != nil {
+			message := fmt.Sprintf("Error extracting field cidr from IPPool %q", pool.GetName())
+			status.QueueFailureMessage(message)
+			continue
+		}
+
+		if !found {
+			message := fmt.Sprintf("No CIDR found in IPPool %q", pool.GetName())
+			status.QueueFailureMessage(message)
+			continue
+		}
+		ippools[cidr] = pool
+	}
+
+	for _, gateway := range gateways.Items {
+		if gateway.Status.HAStatus != submv1.HAStatusActive {
+			continue
+		}
+
+		for _, connection := range gateway.Status.Connections {
+			for _, subnet := range connection.Endpoint.Subnets {
+				ipPool, found := ippools[subnet]
+				if found {
+					isDisabled, err := getValue(ipPool, "disabled")
+					if err != nil {
+						status.QueueFailureMessage(err.Error())
+						continue
+					}
+
+					// When disabled is set to true, Calico IPAM will not assign addresses from this Pool.
+					// The IPPools configured for Submariner remote CIDRs should have disabled as true.
+					if !isDisabled {
+						status.QueueFailureMessage(fmt.Sprintf("IPPool %q for cidr %q has disabled set to false", ipPool.GetName(), subnet))
```

Comment:
Perhaps printing the `Endpoint` name would be helpful too?

```suggestion
						status.QueueFailureMessage(fmt.Sprintf("The IPPool %q with CIDR %q from remote Endpoint %q is disabled", ipPool.GetName(), subnet, connection.Endpoint.CableName))
```